52 54 flipcards

bluethreadmade opened the pull request:
Closes JIRA Issues CT-37, CT-38, CT-39

- refactored TeamGridLayout, moving TeamMemberCard to it's own component
- built TeamMemberCardFront and TeamMemberCardBack components
- added bio type to MemberInfo.ts
- imports react and useState to TeamGridLayout - and added card flip logic to Team function

Expected Function:
- all cards show front when page is loaded
- click 1 card to flip it
- only 1 card is flipped at a time, when a second one is clicked - the first goes back to showing the Front

view 1
![image](https://github.com/user-attachments/assets/cd82aab0-62de-4988-bc16-a18e3ebd744b)

view2 - flipped card
![image](https://github.com/user-attachments/assets/b5584b4f-ea7e-42b8-8ffb-e2c3c5cb8a67)

ec-davis reviewed (commented):
Tests
- [x] all cards show front when page is loaded
- [ ] click 1 card to flip it
- [ ] only 1 card is flipped at a time, when a second one is clicked - the first goes back to showing the Front

ec-davis commented on src/components/TeamGridLayout/index.tsx:
```diff
@@ -1,63 +1,48 @@
+import TeamMemberCardBack from "../TeamMemberCardBack/index";
+
 
 export default function Team() {
```
Decide whether to call it Team or TeamGridLayout

bluethreadmade replied:
got it - will go with TeamGridLayout - as I think we may end up with a team grid layout and a team carousel layout seperately

ec-davis commented on src/components/TeamGridLayout/index.tsx:
```diff
@@ -1,63 +1,48 @@
+import React, { useState } from "react";
```
React is never used, which will throw a build error from Render

bluethreadmade replied:
fixed

ec-davis commented on src/components/TeamGridLayout/index.tsx:
```diff
@@ -1,63 +1,48 @@
 export default function Team() {
+    // State to track the index of the currently flipped card
+    const [flippedCardIndex, setFlippedCardIndex] = useState<number | null>(null);
+
```
I appreciate the comments in here

ec-davis commented on src/components/TeamMemberCard/index.tsx:
```diff
@@ -1,11 +1,69 @@
+import EmailIcon from "@mui/icons-material/Email";
+import LinkedInIcon from "@mui/icons-material/LinkedIn";
 
 interface TeamMemberCardProps {
```
I think this can be replaced by using the MemberInfo type from types/MemberInfo.ts. I'm still new to doing this, so I'm not 100% confident.

bluethreadmade replied:
i don't think the icons can be put in the memberInfo.ts - but I think it should be re-evaluated within the teammemberLinks component for the best method.

this comment did lead me to make some changes though - specifically in 17-19 in teamMemberCardFront - referencing the entire memberInfo as a passed prop instead of each one individually.

ec-davis commented on src/components/TeamMemberCard/index.tsx:
```diff
@@ -1,11 +1,69 @@
+        </Tooltip>
+        <Tooltip title="Email me. But this doesn't work yet. Sadness.">
+          <IconButton
+          // MS Co-Pilot gave me the code below, but doing this on my machine results in an
```
Add the following
"TODO: Implement email link (https://crestonetech.atlassian.net/browse/CT-45)"

bluethreadmade replied:
done

ec-davis commented on src/components/TeamMemberCardBack/index.tsx:
```diff
@@ -0,0 +1,71 @@
+} from "@mui/material";
+
+// import GitHubIcon from "@mui/icons-material/GitHub";
+// import EmailIcon from "@mui/icons-material/Email";
```
Can we delete these lines?

bluethreadmade replied:
yes

ec-davis commented on src/components/TeamMemberCardBack/index.tsx:
```diff
@@ -0,0 +1,71 @@
+  Card,
+  CardContent,
+  Typography,
+//   Tooltip,
```
delete commented lines?

bluethreadmade replied:
yes

ec-davis commented on src/components/TeamMemberCardBack/index.tsx:
```diff
@@ -0,0 +1,71 @@
+// import LinkedInIcon from "@mui/icons-material/LinkedIn";
+
+interface TeamMemberCardBackProps {
+//   memberName: string;
```
Can you remove the commented code?

bluethreadmade replied:
yes

ec-davis reviewed (changes requested):
Overall: awesome, it works!!!

Refining: I added comments about tidying up, e.g. removing commented code

bluethreadmade commented on src/components/TeamGridLayout/index.tsx:
```diff
@@ -1,63 +1,48 @@
+                  // memberName={member.name}
+                  // github={member.github}
+                  // email={member.email}
+                  // linkedIn={member.linkedIn}
```
removing - they will end up in the team links component anyway

bluethreadmade commented on src/components/TeamMemberCardBack/index.tsx:
```diff
@@ -0,0 +1,71 @@
+        <Tooltip title="Email me. But this doesn't work yet. Sadness.">
+          <IconButton
+          // MS Co-Pilot gave me the code below, but doing this on my machine results in an
+          // opening a new browser tab, with the mailto link in the address bar - but the tab
```
done

bluethreadmade commented:
I just re-requested review @ec-davis - I cleaned up all the old commented code, and updated the way that memberInfo was being used, which made the code cleaner and more reusable imho. I also deleted the TeamMemberCard directory - because there isn't a card anymore - there's a front and a back, and one of those gets displayed in the grid based on the flip state.
